src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, PropsWithChildren, ReactNode, useContext, useMemo, useState } from "react";
+import { Toast, ToastTypes } from "./Toast";
+
+export interface ToastNoticeProps {
+  type: ToastTypes;
+  message: ReactNode;
+}
+
+export type ToastContextProps = {
+  notice: ToastNoticeProps | undefined;
+  setNotice: React.Dispatch<React.SetStateAction<ToastNoticeProps | undefined>>;
+};
+
+export const ToastContext = createContext<ToastContextProps>({
+  setNotice: () => {},
+  notice: { type: "error", message: "" },
```

Comment:
I don't think we want to explicitly set `type: error`, but when I tried to add an `undefined/null` type to `ToastTypes`, I then got an error when trying to use the type as an index.